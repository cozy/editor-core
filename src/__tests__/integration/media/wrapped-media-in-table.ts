import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { editable, getDocFromElement, fullpage } from '../_helpers';
import {
  goToEditorTestingWDExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import { waitForNumImages } from './_utils';
import adf from './_fixtures_/wrapped-media-in-table.adf.json';

BrowserTestCase(
  'wrapped-media-in-table.ts: Allows clicking next to wrapped media in table',
  { skip: ['edge', 'safari', 'firefox'] },
  async (
    client: Parameters<typeof goToEditorTestingWDExample>[0],
    testCase: string,
  ) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(adf),
      allowTables: true,
      media: {
        allowMediaSingle: true,
      },
    });
    await waitForNumImages(page, 2);
    await page.moveTo('.mediaSingleView-content-wrap:first-of-type', 180, 30);
    await page.click();
    await page.type(editable, 'hello');
    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testCase);
  },
);
