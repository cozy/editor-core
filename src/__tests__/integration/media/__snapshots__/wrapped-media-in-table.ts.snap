// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapped-media-in-table.ts: Allows clicking next to wrapped media in table 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  382,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  122,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  253,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  382,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "layout": "wrap-left",
                    "width": 15,
                  },
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "__contextId": "DUMMY-OBJECT-ID",
                        "__displayType": null,
                        "__external": false,
                        "__fileMimeType": "image/jpeg",
                        "__fileName": "example.png",
                        "__fileSize": 58705,
                        "alt": null,
                        "collection": "MediaServicesSample",
                        "height": 980,
                        "id": "1234-5678-abcd-efgh",
                        "occurrenceKey": null,
                        "type": "file",
                        "url": null,
                        "width": 840,
                      },
                      "type": "media",
                    },
                  ],
                  "type": "mediaSingle",
                },
                Object {
                  "content": Array [
                    Object {
                      "text": "hello",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
                Object {
                  "attrs": Object {
                    "layout": "center",
                    "width": null,
                  },
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "__contextId": "DUMMY-OBJECT-ID",
                        "__displayType": null,
                        "__external": false,
                        "__fileMimeType": "image/jpeg",
                        "__fileName": "example.png",
                        "__fileSize": 58705,
                        "alt": null,
                        "collection": "MediaServicesSample",
                        "height": 683,
                        "id": "1234-5678-abcd-efgh",
                        "occurrenceKey": null,
                        "type": "file",
                        "url": null,
                        "width": 1024,
                      },
                      "type": "media",
                    },
                  ],
                  "type": "mediaSingle",
                },
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  122,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  253,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  382,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  122,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": Array [
                  253,
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;
